web/src/components/MapView.tsx

```diff
@@ -37,35 +31,23 @@ type MapViewProps = {
   currentFeature: PoiFeature | null
   selectFeature: (feature: PoiFeature | null, restoreScrollPosition: boolean) => void
   changeSnapPoint?: (snapPoint: number) => void
-  cityModel: CityModel
   languageCode: string
   children: ReactNode
+  viewport: MapViewViewport
+  setViewport: (mapViewport: React.SetStateAction<MapViewViewport>) => void
 }
 
 type MapCursorType = 'grab' | 'auto' | 'pointer'
 
-const moveViewToBBox = (bBox: BBox, defaultVp: MapViewMercatorViewport): MapViewMercatorViewport => {
-  const mercatorVp = new WebMercatorViewport(defaultVp)
-  return mercatorVp.fitBounds([
-    [bBox[0], bBox[1]],
-    [bBox[2], bBox[3]],
-  ])
-}
-
 const MapView = ({
   featureCollection,
   selectFeature,
   changeSnapPoint,
   currentFeature,
-  cityModel,
+  viewport,
+  setViewport,
   children,
 }: MapViewProps): ReactElement => {
-  const bboxViewport = useMemo(
```

Comment:
Why did you move the viewport logic?

Reply:
I thought about the situation where one reduces the size of his/her browser window or switches between landscape and horizontal on his/her tablet. Then the viewport will be reset, which does not feel right.
I also thought about moving reverting most of the refactoring and putting the map back into the pois page again, but I still think that generally the current implementation makes more sense.

Reply:
I think it is nice the way it is :)